### PR TITLE
feat(dataworker): Establish pattern for using uba client in dataworker functions

### PR DESCRIPTION
**Pull request by nicholaspai**
- Adds some starter code plus test scaffolding we can use to fully use UBA Client inside Dataworker. 

Proposed structure is to add `UBA_x` functions that we can hot-swap for the normal propose/validate/execute functions once we do the swap over.

Alternatively we can add in a switch similar to the relayer where we look up the `version` in the `ConfigStore` in order to determine when to start evaluating UBA bundles

### CODE REVIEW

File: src/dataworker/index.ts [hunk 1]
```diff
@@ -94,6 +95,15 @@ export async function runDataworker(_logger: winston.Logger, baseSigner: Wallet)
...
       );
 
+      const ubaClient = new UBAClient(config.chainIdListIndices, clients.hubPoolClient, spokePoolClients, logger);
+      // TODO: The following UBA client update updates the hub pool, config store and all spoke pool clients.
+      // However, to construct the spoke pool clients with the correct lookbacks, we need to update the hub pool
+      // client first to figure out how far to look back (based on the latest X validated root bundle end blocks), so
+      // its probably best that we add a method to the UBA client that just calls instantiateUBAFeeCalculator
+      // for each chain/token.
+      //
+      // await ubaClient.update()
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/dataworker/index.ts:105]: Also, wdyt about making `spokePoolClients` a public variable in `UBAClient`? Any reason not to?

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/dataworker/index.ts:105]: The `ubaClient` doesn't necessarily need to call `update` each time. That updates the internal clients. After the first update, the remainder of the functions can be called without subsequent updates

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/dataworker/index.ts:105]: > Also, wdyt about making `spokePoolClients` a public variable in `UBAClient`? Any reason not to?

I was just worrying about encapsulation. I can make them public if wanted

[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/dataworker/index.ts:105]: @james-a-morris What I'm trying to say is that I would like a way to trigger the following logic without re-updating the internal clients:

```ts
// Update the UBAFeeCalculators
    await Promise.all(
      Object.entries(this.spokeUBAFeeCalculators).flatMap(([chainId, spokeUBAFeeCalculator]) =>
        Object.keys(spokeUBAFeeCalculator).map((token) =>
          this.instantiateUBAFeeCalculator(parseInt(chainId), token, this.hubPoolClient.latestBlockNumber ?? 0)
        )
      )
    );
```

And yeah can we make `spokePoolClients` public?

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/dataworker/index.ts:105]: Opened a PR in the SDK-V2 that should accomplish your use-case

File: test/mocks/MockUBAClient.ts [hunk 2]
```diff
@@ -1,20 +1,28 @@
-import { clients as sdkClients } from "@across-protocol/sdk-v2";
+import * as UBAClientTypes from "@across-protocol/sdk-v2/src/clients/UBAClient/UBAClientTypes";
+import { UBAActionType } from "@across-protocol/sdk-v2/src/UBAFeeCalculator/UBAFeeTypes";
```
Comment (pxrl) [inline test/mocks/MockUBAClient.ts:2]: I'd prefer to fix the exports upstream rather than be poking down into the SDK-v2 tree.

Comment (nicholaspai) [reply test/mocks/MockUBAClient.ts:2]: yep. https://github.com/across-protocol/sdk-v2/pull/289

**Pull request merged**